indra/newview/llselectmgr.cpp

```diff
@@ -1938,6 +1938,12 @@ void LLSelectMgr::selectionSetGLTFMaterial(const LLUUID& mat_id)
             LLUUID asset_id = mMatId;
             if (mItem)
             {
+                // If success, the material may be copied into the object's inventory
```

Comment:
Looks got to me, but couple questions:
1. Instead of doing this check here, shouldn't it be better to extend allowOperationBy check below (1957) to cover "no transfer"?
2 selectionSetImage handles a case for "te == -1", my assumtion that this never happens and thus isn't covered here? (and actually shouldn't be covered in selectionSetImage)

Reply:
1. Now that I look closer, I see a potential for cleanup here. I think `handleDropMaterialProtections` handles the important edge cases, so probably best to rely exclusively on that. I will push an update if the code works after cleanup.
2. You're probably right that the -1 case is never handled

Reply:
Looking closer at this, I think that this cleanup is not possible. Only thing that seems doable is removing the -1 case. Other cases may still be needed. Probably not worth messing with too much at this testing stage.